plugins/modules/nvme_host.py

```diff
@@ -0,0 +1,484 @@
+
+        try:
+            filter_field = None
+            filter_value = None
+
+            if id_name:
+                filter_field, filter_value = id_name
+
+            all_host_details = self.powerflex_conn.sdc.get()
+            # Assign names to unnamed NVMe hosts and find the target host
+            for nvme_host in all_host_details:
+                if nvme_host.get("name") is None:
+                    nvme_host["name"] = f"NVMeHost:{nvme_host['id']}"
+
+            if filter_field is None or filter_value is None:
+                return all_host_details[0] if all_host_details else None
+
+            nvme_host_details = [
+                host_entity for host_entity in all_host_details
+                if host_entity.get(filter_field) == filter_value and host_entity.get('hostType') == 'NVMeHost'
+            ]
+
+            return nvme_host_details[0] if nvme_host_details else None
+
+        except Exception as e:
+            errormsg = "Failed to get the NVMe host %s with error %s" % (
+                id_name,
+                str(e),
+            )
+            LOG.error(errormsg)
+            self.module.fail_json(msg=errormsg)
+
+    def create_nvme_host(self, nqn, name, max_num_paths, max_num_sys_ports):
+        """Create the NVMe host
+        :param nqn: The NQN of the NVMe host
+        :param name: The name of the NVMe host
+        :param max_num_paths: The maximum number of paths per volume
+        :param max_num_sys_ports: Maximum Number of Ports Per Protection Domain
+        """
+        try:
+            if nqn is None or len(nqn.strip()) == 0:
+                self.module.fail_json(msg="Please provide valid NQN.")
+            host_id = self.powerflex_conn.host.create(
+                nqn=nqn,
+                name=name,
+                max_num_paths=max_num_paths,
+                max_num_sys_ports=max_num_sys_ports,
+            )
+            return host_id
+
+        except Exception as e:
+            errormsg = "Create NVMe host operation failed with " "error %s" % str(e)
+            LOG.error(errormsg)
+            self.module.fail_json(msg=errormsg)
+
+    def validate_parameters(self):
+        """Validate the input parameters"""
+
+        host_identifiers = ["nvme_host_name", "nqn"]
+        for param in host_identifiers:
+            if (
+                self.module.params[param] is not None
+                and len(self.module.params[param].strip()) == 0
+            ):
+                msg = f"Please provide valid {param}"
+                LOG.error(msg)
+                self.module.fail_json(msg=msg)
+
+        if (
+            self.module.params["nvme_host_name"] is None
+            and self.module.params["nqn"] is None
+        ):
+            msg = "Please provide at least one of nvme_host_name or nqn"
+            LOG.error(msg)
+            self.module.fail_json(msg=msg)
+
+    def remove(self, nvme_id):
```

Comment:
Use proper name to identify the method.
```suggestion
    def remove_nvme(self, nvme_id):
```

Reply:
Done